app/src/backend/permissions/ResourceLoader.ts

```diff
@@ -0,0 +1,89 @@
+import { db } from "@/models";
+import type { PermissionContext } from "./PermissionTypes";
+
+/**
+ * Handles loading resources based on permission context
+ */
+export class ResourceLoader {
+  
+  /**
+   * Get resource based on context with appropriate includes
+   */
+  static async getResource(context: PermissionContext): Promise<any> {
+    if (context.inventoryId) {
+      return this.getInventoryWithFullContext(context.inventoryId);
+    }
+    
+    if (context.cityId) {
+      return this.getCityWithContext(context.cityId);
+    }
+    
+    if (context.projectId) {
+      return this.getProjectWithContext(context.projectId);
+    }
+    
+    if (context.organizationId) {
+      return this.getOrganization(context.organizationId);
+    }
+    
+    return null;
+  }
+
+  /**
+   * Load inventory with full hierarchy context
+   */
+  private static async getInventoryWithFullContext(inventoryId: string) {
```

Comment:
This method name is a bit unclear in how it differentiates to the next (full context vs. context).